adafruit_avrprog.py

```diff
@@ -299,14 +307,14 @@ def begin(self, clock=_FAST_CLOCK):
         self._spi.configure(baudrate=clock)
         self._transaction((0xAC, 0x53, 0, 0))
 
-    def end(self):
+    def end(self) -> None:
         """
         End programming mode: SPI is released, and reset pin set high.
         """
         self._spi.unlock()
         self._rst.value = True
 
-    def read_signature(self):
+    def read_signature(self) -> list:
```

Comment:
We can also define the contents of a list using generics.  For lists, we only need to define what any given element can be.  The return from `_transaction()` is `bytearray`, and an element of a `bytearray` is an `int`.  Since we're appending only `int` to the list, we can import `List` from `typing and type the return here as:

```suggestion
    def read_signature(self) -> List[int]:
```